### PR TITLE
MAINT: Refactor more compat checks into compat init

### PR DESCRIPTION
**Pull request by gfyoung**
Title is self-explanatory.

xref #309.

### CODE REVIEW

File: pandas_datareader/compat/__init__.py [hunk 1]
```diff
@@ -1,11 +1,22 @@
...
-pandas_version = LooseVersion(pd.__version__)
-if pandas_version <= '0.19.2':
+PANDAS_VERSION = LooseVersion(pd.__version__)
+
+PANDAS_0190 = (PANDAS_VERSION >= LooseVersion('0.19.0'))
+PANDAS_0170 = (PANDAS_VERSION >= LooseVersion('0.17.0'))
+PANDAS_0160 = (PANDAS_VERSION >= LooseVersion('0.16.0'))
+
+if PANDAS_0190:
+    from pandas.api.types import is_number
```
Comment (jreback) [inline pandas_datareader/compat/__init__.py]: this should be >= 0.19.0

**Comment by jreback** [comment]
minor change ping when ready

File: pandas_datareader/compat/__init__.py (hunk 1 above)
Comment (gfyoung) [inline pandas_datareader/compat/__init__.py]: You do see how PANDAS_0190 is defined?

**Comment by gfyoung** [comment]
> minor change ping when ready

I already addressed in this diff (see how I defined `PANDAS_0190`)

**Comment by jreback** [comment]
ahh i c
ok then

**Comment by jreback** [comment]
thanks!

ideally can get the tests working (or maybe xfail the ones that are not)

**Comment by gfyoung** [comment]
I think #303 is intending to do that, but I'm not sure whether or not progress will be made yet to actually fix the tests themselves. :/

**Pull request merged**